shared-lib/spiutils/src/protocol/firmware.rs

```diff
@@ -322,3 +437,104 @@ impl ToWire for WriteChunkResponse {
         Ok(())
     }
 }
+
+// ----------------------------------------------------------------------------
+
+wire_enum! {
+    /// When to perform the reboot.
+    pub enum RebootTime: u8 {
+        /// Unknown result type.
+        Unknown = 0xff,
+
+        /// Reboot immediately
+        Immediate = 0x00,
+
+        /// Reboot after a delay or when the BMC resets.
+        Delayed = 0x01,
+    }
+}
+
+/// A parsed reboot request.
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub struct RebootRequest {
+    /// When to reboot.
+    pub time: RebootTime,
+}
+
+/// The length of a reboot request on the wire, in bytes.
+pub const REBOOT_REQUEST_LEN: usize = 1;
+
+impl Message<'_> for RebootRequest {
+    const TYPE: ContentType = ContentType::RebootRequest;
+}
+
+impl<'a> FromWire<'a> for RebootRequest {
+    fn from_wire<R: Read<'a>>(mut r: R) -> Result<Self, FromWireError> {
+        let time_u8 = r.read_be::<u8>()?;
+        let time = RebootTime::from_wire_value(time_u8).ok_or(FromWireError::OutOfRange)?;
+        Ok(Self {
+            time,
+        })
+    }
+}
+
+impl ToWire for RebootRequest {
+    fn to_wire<W: Write>(&self, mut w: W) -> Result<(), ToWireError> {
+        w.write_be(self.time.to_wire_value())?;
+        Ok(())
+    }
+}
+
+// ----------------------------------------------------------------------------
+
+wire_enum! {
+    /// The result of a reboot request.
+    pub enum RebootResult: u8 {
```

Comment:
Maybe an Option?

Reply:
Actually, we don't need or want the `Unknown` value. I removed it. Good catch!